### PR TITLE
Add type definitions for ghauth 3.2

### PR DESCRIPTION
**Pull request by Leko**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/ghauth/ghauth-tests.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import ghauth, { AuthOptions, TokenData } from "ghauth";
```
Comment (amcasey) [inline types/ghauth/ghauth-tests.ts]: Default ES6 imports aren't callable.  Seems like you might have wanted
```ts
import ghauth = require("ghauth");
import { AuthOptions, TokenData } from "ghauth";
```

File: types/ghauth/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+     */
+    userAgent?: string;
+}
+
+export interface TokenData {
+    user: string;
+    token: string;
+}
+
+export default function ghauth(authOptions: AuthOptions, callback: (err: Error, tokenData: TokenData) => void): void;
```
Comment (amcasey) [inline types/ghauth/index.d.ts]: Seems like this should be
```ts
function ghauth…
export = ghauth;
```

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/ghauth/ghauth-tests.ts (hunk 1 above)
Comment (Leko) [reply types/ghauth/ghauth-tests.ts]: Thank you for your review.
Done.

File: types/ghauth/index.d.ts (hunk 2 above)
Comment (Leko) [reply types/ghauth/index.d.ts]: Revise done.

**Review by amcasey — APPROVED** [review]

File: types/ghauth/ghauth-tests.ts (hunk 1 above)
Comment (amcasey) [reply types/ghauth/ghauth-tests.ts]: Thanks!

**Pull request merged**